### DISCUSSION TITLE
How to ignore pytest warning logs?

### DISCUSSION THREAD

**Discussion by Pranjilagrawal** [pytest-dev/pytest#9853] [Q&A]
Hello, I need one help,i have searched everywhere,tried all the workarounds but no luck.
I get this warning message "**[WARNING] connectionpool.py:303 Connection pool is full, discarding connection: 127.0.0.1**" in my logs.
For other warnings i use this piece of code in **pytest.ini** file --
```
filterwarnings =
   ignore::UserWarning
   error
   ignore:function ham\(\) is deprecated:DeprecationWarning
   ignore::DeprecationWarning
```
**Can someone help me what i should use to get rid of above warning message?**
Thanks in advance!

**Comment by portyu9** [comment]
Hi @Pranjilagrawal 

That message is a **Python logging record**, not a Python `warnings.warn()` warning, so `filterwarnings` will not affect it.

It comes from the `urllib3.connectionpool` logger.

If you simply do not want pytest to show that logger, current pytest supports:

```bash
pytest --log-disable=urllib3.connectionpool
```

Or configure the logger itself:

```python
import logging

logging.getLogger("urllib3.connectionpool").setLevel(logging.ERROR)
```

That suppresses its WARNING-level messages while leaving other loggers alone.

I would also check why the connection pool is becoming full if that is unexpected, because suppressing the log only hides the symptom; it does not change urllib3's connection-pool behavior.

Please mark this answer as accepted if it helped, thank you.
